src/ddo/interfaces/Service.ts

```diff
@@ -1,12 +1,27 @@
 import { Metadata } from './Metadata'
 import { Status } from './Status'
 
-export type ServiceType = 'authorization' | 'metadata' | 'access' | 'compute'
+export interface CustomData {
+  name: string
+  type: string
+  label: string
+  required: boolean
+  options?: any
+  description: string
+}
 
+export interface RequiredData {
+  userdata?: CustomData[]
+  algodata?: CustomData[]
+}
+
+export type ServiceType = 'authorization' | 'metadata' | 'access' | 'compute'
 export interface ServiceCommonAttributes {
   main: { [key: string]: any }
   additionalInformation?: { [key: string]: any }
   status?: Status
+  userdata?: CustomData[]
```

Comment:
duplication, `ServiceCommonAttributes` can extend `RequiredData` so we do not have to repeat this here:

```ts
export interface ServiceCommonAttributes extends RequiredData {
  ...
}
```